app/controllers/provider/sessions_controller.rb

```diff
@@ -7,9 +7,10 @@ class Provider::SessionsController < FrontendController
   before_action :ensure_provider_domain
   before_action :find_provider
   before_action :instantiate_sessions_presenter, only: [:new, :create]
+  before_action :redirect_if_logged_in, :only => [:new]
```

Comment:
```suggestion
  before_action :redirect_if_logged_in, only: %i[new]
```

I see that the method `redirect_if_logged_in` already exists in this file and contains:
```ruby
def redirect_if_logged_in
  if logged_in? && current_account.provider?
    redirect_to provider_admin_dashboard_path
  end
end
```
I wonder if the condition `current_account.provider?` is necessary... is this controller ever used for buyers? I think not, but if so, what do we have to do when the logged in account is a buyer?